packages/core/saved-objects/core-saved-objects-server/src/extensions/security.ts

```diff
@@ -209,19 +262,49 @@ export interface RedactNamespacesParams<T, A extends string> {
   typeMap: AuthorizationTypeMap<A>;
 }
 
+export interface AuthorizeCreateObject {
+  type: string;
+  id: string;
+  initialNamespaces: string[] | undefined;
+  existingNamespaces: string[] | undefined;
+}
+
 /**
  * The ISavedObjectsSecurityExtension interface defines the functions of a saved objects repository security extension.
  * It contains functions for checking & enforcing authorization, adding audit events, and redacting namespaces.
  */
 export interface ISavedObjectsSecurityExtension {
+  // ToDo: Document
+  // authorizeCreate: <T extends string>(
+  //   action: SecurityAction.CREATE | SecurityAction.BULK_CREATE,
+  //   namespaceString: string,
+  //   objects: Array<{
+  //     type: string;
+  //     id: string;
+  //     initialNamespaces: string[] | undefined;
+  //     existingNamespaces: string[] | undefined;
+  //   }>
+  // ) => Promise<CheckAuthorizationResult<T> | undefined>;
+
+  /**
+   * Performs authorization for create actions
+   * @param params - actions, types & spaces map, audit callback, options (enforce bypassed if enforce map is undefined)
+   * @returns CheckAuthorizationResult - the resulting authorization level and authorization map
```

Comment:
I will actually eliminate the possibility of an undefined return - it will be known which actions do not require a check (thus no result), so we can simplify this.

Reply:
That's great!